iotile_analytics_core/iotile_analytics/core/session.py

```diff
@@ -131,10 +132,19 @@ def __init__(self, user=None, password=None, domain=DOMAIN_NAME, verify=None, to
             return
 
         if user is None:
-            user = input("Please enter your IOTile.cloud email: ".encode('utf-8'))
+            # Both python 2 and 3 require native strings to be passed into getpass
+            prompt_str = "Please enter your IOTile.cloud email: "
+            if sys.version_info.major < 3:
+                prompt_str = prompt_str.encode('utf-8')
```

Comment:
Technically it shouldn't be because `input` expects a native `str`, which on python 2 is not unicode but for python 3 compatibility it's better to set `from __future__ import unicode_literals` which then breaks this function sometimes depending on the underlying console its interacting with.  

However, python 3 requires that you give it a `str` and not `bytes` so encoding it wrong on python 3.